### PR TITLE
Add tools test for buildWeb compilation

### PR DESCRIPTION
**Pull request by jmagman**
Add some baseline web compilation tool tests to make changes like https://github.com/flutter/flutter/pull/124165 easier to test.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter_tools/lib/src/web/compile.dart [hunk 1]
```diff
@@ -16,96 +16,112 @@ import '../globals.dart' as globals;
...
-  ];
+  final Logger _logger;
+  final BuildSystem _buildSystem;
+  final Usage _flutterUsage;
+  final FlutterVersion _flutterVersion;
+  final FileSystem _fileSystem;
 
-  final ProjectMigration migration = ProjectMigration(migrators);
-  migration.run();
+  Future<void> buildWeb(
```
Comment (jmagman) [inline packages/flutter_tools/lib/src/web/compile.dart:44]: Most of this method is whitespace changes, which you can optionally ignore in the GitHub UI for this change, if you want:
Screenshot 2023-04-04 at 7 10 32 PM

File: packages/flutter_tools/test/general.shard/web/compile_web_test.dart [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    });
+
+    final WebBuilder webBuilder = WebBuilder(
+      logger: logger,
+      buildSystem: buildSystem,
+      usage: testUsage,
+      flutterVersion: flutterVersion,
+      fileSystem: fileSystem,
+    );
+    await webBuilder.buildWeb(
```
Comment (jmagman) [inline packages/flutter_tools/test/general.shard/web/compile_web_test.dart:77]: This covers only one permutations of these parameters, but hopefully this gives an example that can be expanded on by the web team in the future.

**Review by kevmoo — APPROVED** [review]
Love it! Thanks!

CC @yjbanov

**Review by eyebrowsoffire — APPROVED** [review]
This is awesome! Thank you!

**Pull request merged**